plugin.video.showcase.canada/lib/scraper.py

```diff
@@ -0,0 +1,158 @@
+   pid = 'http://data.media.theplatform.com/media/data/Category/%s' % geurl
+   wewait = True
+   for b in a:
+      if b['parentId'] == pid:
+          name = b['title']
+          url  = b['id'].rsplit('/',1)[1]
+          ilist = self.addMenuItem(name,'GE', ilist, url, self.addonIcon, self.addonFanart, {}, isFolder=True)
+   return(ilist)
+
+
+  def getAddonEpisodes(self,url,ilist):
+    addonLanguage  = self.addon.getLocalizedString
+    self.defaultVidStream['width']  = 640
+    self.defaultVidStream['height'] = 360
+    geurl = uqp(url)
+    url = 'http://feed.theplatform.com/f/dtjsEC/9H6qyshBZU3E?count=true&byCategoryIDs=%s&startIndex=1&endIndex=100&sort=pubDate|desc&callback=' % geurl
+    html  = self.getRequest(url)
+    a = json.loads(html)['entries']
+    if len(a) == 0:
+     url = 'http://common.farm1.smdg.ca/Forms/PlatformVideoFeed?platformUrl=http%3A//feed.theplatform.com/f/dtjsEC/9H6qyshBZU3E/categories%3Fpretty%3Dtrue%26byHasReleases%3Dtrue%26byCustomValue%3D%7Bplayertag%7D%7Bz/Showcase%20Video%20Centre%7D%26sort%3DfullTitle&callback='
+     html = self.getRequest(url)
+     a = json.loads(html[1:len(html)-1])['items']
+     pid = 'http://data.media.theplatform.com/media/data/Category/%s' % geurl
+     wewait = True
+     for b in a:
+      if b['parentId'] == pid:
+        if wewait == True:
+          if b['title'] == 'Full Episodes': 
+             wewait = False
+             pid = b['id']
+          if b['hasReleases'] == False: continue
+        if wewait == False and b['hasReleases'] == True:
+          name = b['title']
+          url  = b['id'].rsplit('/',1)[1]
+          ilist = self.addMenuItem(name,'GE', ilist, url, self.addonIcon, self.addonFanart, {}, isFolder=True)
+
+    else:  
+     for b in a:
+      url = b['content'][0]['url']
+      name    = h.unescape(b['title'])
+      thumb   = b['defaultThumbnailUrl']
+#      fanart  = self.addonFanart
+      fanart = thumb
+      infoList = {}
+      infoList['Duration']    = int(b['content'][0]['duration'])
+      infoList['Title']       = name
+      try: infoList['Studio']      = b['pl1$network']
+      except: pass
+      infoList['Date']        = datetime.datetime.fromtimestamp(b['pubDate']/1000).strftime('%Y-%m-%d')
+      infoList['Aired']       = infoList['Date']
+      infoList['Year']        = int(infoList['Date'].split('-',1)[0])
+      try:    infoList['MPAA'] = re.compile('ratings="(.+?)"',re.DOTALL).search(html).group(1).split(':',1)[1]
+      except: infoList['MPAA'] = None
+      try:    infoList['Episode'] = int(b['pl1$episode'])
+      except: infoList['Episode'] = None
+      try:    infoList['Season']  = int(b['pl1$season'])
+      except: infoList['Season']  = 1
+      infoList['Plot']        = h.unescape(b["description"])
+      infoList['TVShowTitle'] = b['pl1$show']
+      ilist = self.addMenuItem(name,'GV', ilist, url, thumb, fanart, infoList, isFolder=False)
+    return(ilist)
+
+  def getProxyRequest(self, url):
```

Comment:
Do you mean getProxyRequest? I've been thinking about just adding a "use_proxy" parm and add the logic into getRequest(). Haven't got there yet though.